core/src/dird/backup.cc

```diff
@@ -945,16 +945,18 @@ void GenerateBackupSummary(JobControlRecord *jcr, ClientDbRecord *cr, int msg_ty
       }
    }
 
-   if (jcr->ReadBytes == 0) {
+   if (jcr->ReadBytes == 0 || !FindUsedCompressalgos(&compress_algo_list, jcr)) {
+     // compress_algo_list is guaranteed to be emtpy
       bstrncpy(compress, "None", sizeof(compress));
    } else {
-      compression = (double)100 - 100.0 * ((double)jcr->JobBytes / (double)jcr->ReadBytes);
-      if (compression < 0.5) {
-         bstrncpy(compress, "None", sizeof(compress));
-      } else {
-         Bsnprintf(compress, sizeof(compress), "%.1f %%", compression);
-         FindUsedCompressalgos(&compress_algo_list, jcr);
-      }
+     double compression = 100.0 - 100.0 * ((double)jcr->JobBytes / (double)jcr->ReadBytes);
+     if (compression < -1 && jcr->is_JobLevel(L_FULL)) {
+       Jmsg(jcr, M_INFO, 0,
+	    _("Compression inflated the full backup data by more than 1%%."
+	      " We suggest to use a different algorithm or to disable "
+	      "compression.\n"));
```

Comment:
This needs to be fixed for it to be mergable.
```suggestion
	    T_("Compression inflated the full backup data by more than 1%%."
	       " We suggest to use a different algorithm or to disable "
	       "compression.\n"));
```